Migrate to uv

ulgens opened the pull request:
Resolves https://github.com/django/djangoproject.com/issues/1901

Depends on:
- https://github.com/django/djangoproject.com/pull/2448

ulgens commented:
@django/django-website Is there a way to disable pre-commit.ci?

pauloxnet commented:
> ... I still need to understand why black is generating the trailing commas ...

Do we still use black? Maybe it's time to migrate linting and formatting to ruff :smile:

ulgens commented:
> Do we still use black? Maybe it's time to migrate linting and formatting to ruff 😄

+100 🌻 I was a bit afraid of starting that discussion but there is no harm in creating an issue.

ulgens commented:
After the last review from @pauloxnet , I pushed some changes:
- Rebased from main after #2448 , no more mysterious trailing commas
- Moved pyproject-fmt settings to pyproject.toml file
- Replaced the ulgens/uv-lock-check githook with astral-sh/uv-pre-commit
- Updated makefile content and wrapped the necessary entries with `uv run`

ulgens commented:
> Do you think that using the iterative approach will help everyone to review and split the complexity of this PR

I can try that 👍🏻 

> P.S. avoid force pushing changes only because it' more difficult to see what has been changed since last review :)

I prefer push force - rebase to keep the PR history clean but I see your point, will be more careful with them.
